src/tiledquickplugin/tilelayeritem.cpp

```diff
@@ -173,6 +174,206 @@ static void drawTileLayer(QSGNode *parent,
         parent->appendChildNode(new TilesNode(helper.texture(), tileData));
 }
 
+// Avoids passing lots of variables to static free functions.
+class IsometricRenderHelper
+{
+public:
+    IsometricRenderHelper(Tiled::MapRenderer *renderer,
+                          QSGNode *parent,
+                          const MapItem *mapItem,
+                          const TileLayer *layer,
+                          const QRect &rect) :
+        mRenderer(renderer),
+        mParent(parent),
+        mMapItem(mapItem),
+        mLayer(layer),
+        mRect(rect),
+        mTilesetHelper(mapItem),
+        mTilesWide(rect.width()),
+        mTilesHigh(rect.height()),
+        mTileWidth(mapItem->map()->tileWidth()),
+        mTileHeight(mapItem->map()->tileHeight()),
+        mTilesCreated(0),
+        mStartPos(QPoint((rect.width() * mTileWidth) / 2, 0))
+    {
+    }
+
+    void addTilesToNode();
+
+private:
+    QPoint indexToMapPos(int index) const;
+    void appendTileData(int index);
+
+    Tiled::MapRenderer *mRenderer;
+    QSGNode *mParent;
+    const MapItem *mMapItem;
+    const TileLayer *mLayer;
+    const QRect mRect;
+    TilesetHelper mTilesetHelper;
+    const int mTilesWide;
+    const int mTilesHigh;
+    const int mTileWidth;
+    const int mTileHeight;
+    QVector<TileData> mTileData;
+    int mTilesCreated;
+    QPoint mStartPos;
+};
+
+QPoint IsometricRenderHelper::indexToMapPos(int index) const
+{
+    return QPoint((index % mTilesWide), qFloor(index / mTilesWide));
+}
+
+void IsometricRenderHelper::appendTileData(int index)
+{
+    const QPoint mapPos = indexToMapPos(index);
+    const Cell &cell = mLayer->cellAt(mapPos.x(), mapPos.y());
+    if (cell.isEmpty())
+        return;
+
+    Tileset *tileset = cell.tile()->tileset();
```

Comment:
You get the tileset by doing `cell.tileset()`, because `cell.tile()` may return null when no tile with the `cell.tileId()` is found in the tileset (maybe I should call it `cell.findTile()`...)